### PR TITLE
feat: add Link component

### PR DESCRIPTION
**Pull request by jaredcwhite**
## Issue Overview

This PR addresses #24 

## Description

Adds a `<Link>` component which can be used throughout an application (instead of, say, directly using Next's `<Link>` component). In addition, in a future scenario where there's a prose section, adds an underlined link style there too.

## How Can This Be Tested/Reviewed?

Storybook: https://deploy-preview-48--storybook-ui-seeds.netlify.app/?path=/story/actions-link--basic-link

## Checklist:

- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have exported any new components
- [x] My commit message(s) and PR title are polished in the conventional commit format, and any breaking changes are indicated in the message and are well-described

### CODE REVIEW

File: src/actions/__stories__/Link.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+export const externalLink = () => (
+  <div style={{ display: "flex", gap: "1rem" }}>
+    <Link href="/test">Internal Link</Link>
+    <Link href="https://www.exygy.com">External Link</Link>
+  </div>
+)
+
+export const linksWithIcons = () => (
+  <div style={{ display: "flex", gap: "1rem" }}>
+    <Link leadIcon={<Icon icon={faHeart} />}>Lead Icon</Link>
```
Comment (ludtkemorgan) [inline src/actions/__stories__/Link.stories.tsx:31]: Without the `href` the link doesn't work, it is just underlined text. I'm wondering if `href` should be a required prop in the component

**Comment by jaredcwhite** [comment]
@slowbot Ah, good to know. I'll update this and the Button version of links

**Comment by emilyjablonski** [comment]
@slowbot One of the links you shared indicates WCAG [recommends not opening links in a new tab](https://codersblock.com/blog/external-links-new-tabs-and-accessibility/#accessibility-implications) for a11y - any thoughts on that?

**Comment by jaredcwhite** [comment]
@emilyjablonski Hmm, I wonder if it's context specific? [That W3C notice](https://www.w3.org/TR/WCAG20-TECHS/G200.html) contains the following about when it's better to open a new tab/window:

> Opening a page containing context-sensitive information, such as help instructions, or an alternate means of completing a form, such as a calendar-based date picker, will significantly disrupt a multi-step workflow, such as filling in and submitting a form, if the page is opened in the same window or tab.

So…maybe if you're on a listing and you click an apply button/link to an external website, it _shouldn't_ open a new window because it's the next logical step…but if you're in the middle of a form or search process and there's an external link to some more context, it _should_ open a new window?

**Comment by emilyjablonski** [comment]
^ That sounds perfect! Agreed

**Comment by jaredcwhite** [comment]
All right, I've updated the Button & Link components with a new `newWindowTarget` prop with code comment "Set external link target if it meets accessibility criteria" — presumably we can include that accessibility criteria for when it should load a new window or not in the zeroheight docs.

**Review by emilyjablonski — APPROVED** [review]

**Review by ludtkemorgan — APPROVED** [review]
🔗

**Pull request merged**